lib/commands/run.coffee

```diff
@@ -135,22 +135,27 @@ maybeInspectContainer = (docker, name) ->
 # Helper method to check and see if the running container's links differ from the ones we would
 # want to create it with. If they do differ, the only recourse is to remove and recreate the
 # container, since you may not (currently) modify links of a running container.
-isLinkMissing = (containerInfo, createOpts) ->
-  # These are the links as reported back by Docker, which have the format:
-  #  /sourceContainer:/destContainer/alias
-  # We do a bit of splitting and joining to convert them back to /sourceContainer:alias so that
-  # we can compare them directly with the Links parameter we provide to create.
-  currentLinks = _.map (containerInfo?.HostConfig?.Links or []), (link) ->
-    [source, dest] = link.split(':')
-    "#{source}:#{dest.split('/').pop()}"
+isLinkMissing = (containerInfo, createOpts, servicesConfig) ->
+  service = containerInfo.Name.split("/")[1].split(".")[0]
+  env = containerInfo.Name.split("/")[1].split(".")[1]
+  prereqs = ServiceHelpers.generatePrereqServices(service, servicesConfig)
```

Comment:
Ah good point, we can rely on the order that containers get started (since transitive dependencies will be started first)